src/common/transformations/src/transformations/init_node_info.cpp

```diff
@@ -8,20 +8,36 @@
 #include <vector>
 
 #include "itt.hpp"
+#include "openvino/op/util/op_types.hpp"
 #include "openvino/op/util/sub_graph_base.hpp"
+#include "transformations/rt_info/constant_node.hpp"
 #include "transformations/rt_info/fused_names_attribute.hpp"
 #include "transformations/rt_info/primitives_priority_attribute.hpp"
 #include "transformations/utils/utils.hpp"
 
 bool ov::pass::InitNodeInfo::run_on_model(const std::shared_ptr<ov::Model>& f) {
     RUN_ON_FUNCTION_SCOPE(InitNodeInfo);
 
-    for (auto& node : f->get_ops()) {
+    for (auto& node : f->get_ordered_ops()) {
         // Recursively apply transformation for sub-graph based operations
         ov::op::util::process_subgraph(*this, node);
 
         auto& rtInfo = node->get_rt_info();
         rtInfo.emplace(FusedNames::get_type_info_static(), FusedNames{node->get_friendly_name()});
+        if (ov::op::util::is_constant(node)) {
+            mark_as_constant_node(node);
+        } else {
+            const auto in_size = node->input_values().size();
+            size_t in_idx = 0;
+            while (in_idx < in_size) {
+                if (!is_marked_as_constant_node(node->get_input_node_shared_ptr(in_idx))) {
+                    break;
+                }
+                if (++in_idx == in_size) {
+                    mark_as_constant_node(node);
+                }
+            }
```

Comment:
Please note that some nodes can be non-const even if all inputs are constants. For example, `op::v8::RandomUniform`, `op::v14::Inverse`, `op::v13::Multinomial` - this is a set of always non-const ops from opset used in CPU.
It seems like we need cover such nodes separately.